ostd/src/cpu/cpu_local.rs

```diff
@@ -170,6 +182,19 @@ impl<T> Deref for CpuLocalDerefGuard<'_, T> {
     }
 }
 
+/// Sets the base address of the CPU-local storage for the bootstrap processor.
+///
+/// It should be called early to let [`crate::task::disable_preempt`] work,
+/// which needs to update a CPU-local preempt lock count. Otherwise it may
+/// panic when calling [`crate::task::disable_preempt`].
+pub(crate) fn early_init_bsp_local_base() {
```

Comment:
This is not a safe function, but as you said that most initialization functions are not safe. So I'm not sure what's the right way to deal with this problem.

Reply:
Ur right, it must be called on the BSP, as a safety condition.